### DISCUSSION TITLE
Firestore field projection - return a subset of all fields

### DISCUSSION THREAD

**Discussion by aemelyanovff** [firebase/flutterfire#9547] [Feature request] [answered]
Firestore supports returning only a subset of all fields. By choosing only the fields that I need I can reduce the amount of data transferred.

References:
* [Rest API/Projection](https://firebase.google.com/docs/firestore/reference/rest/v1/StructuredQuery#projection)
* [`select` in the NodeJS client](https://cloud.google.com/nodejs/docs/reference/firestore/latest/firestore/query#_google_cloud_firestore_Query_select_member_1_)

**Comment by rrousselGit** [comment, marked as answer]
It doesn't appear to be an API available in the client SDKs. This looks to be an admin only API which won't be available for FlutterFire.
